### PR TITLE
Nesthub: init at 1.2.0

### PR DESCRIPTION
**Pull request by nateinaction**
###### Motivation for this change
Create a NixOS package and service to bridge Nest thermostats and Apple HomeKit.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/misc/nesthub.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to enable the Nesthub service.";
+      };
+      configPath = mkOption {
+        type = types.str;
+        default = "/var/lib/nesthub/config.json";
+        description = "Path to the config file.";
+      };
```
Comment (legendofmiracles) [inline nixos/modules/services/misc/nesthub.nix:19]: I don't know what nesthub is, nor how it works - but is the user expected to write their own config file to that path, or does nesthub generate one and place it there? (you have to configure it through the application)

File: nixos/modules/services/misc/nesthub.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+let
+  cfg = config.services.nesthub;
+in {
+  options = {
+    services.nesthub = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to enable the Nesthub service.";
+      };
```
Comment (legendofmiracles) [inline nixos/modules/services/misc/nesthub.nix:14]:
```suggestion
      enable = mkEnableOption "Whether to enable the Nesthub service.";
```

File: nixos/modules/services/misc/nesthub.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        description = "Path to the config file.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.nesthub = {
+      description = "Nesthub Homekit bridge for Nest thermostats";
+      serviceConfig = {
+        ExecStart = "${pkgs.nesthub}/bin/nesthub -config ${config.services.nesthub.configPath}";
```
Comment (legendofmiracles) [inline nixos/modules/services/misc/nesthub.nix:27]:
```suggestion
        ExecStart = "${pkgs.nesthub}/bin/nesthub -config ${cfg.configPath}";
```

File: nixos/modules/services/misc/nesthub.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      };
+      after = [ "network.online" ];
+      wantedBy = [ "multi-user.target" ];
+    };
+  };
+}
+
```
Comment (legendofmiracles) [inline nixos/modules/services/misc/nesthub.nix:34]:
```suggestion
```

File: pkgs/servers/misc/nesthub/default.nix [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  vendorSha256 = "1hm97iyj7cwh7vckj3sv0q93wp1bxjj24bjkrzpphgas7v559dj9";
+
+  meta = with lib; {
+    description = "HomeKit bridge for Nest thermostats";
+    homepage = "https://github.com/yangl1996/nesthub";
+    license = licenses.mit;
+    maintainers = with maintainers; [ nateinaction ];
+  };
+}
+
```
Comment (legendofmiracles) [inline pkgs/servers/misc/nesthub/default.nix:23]:
```suggestion
```

**Review by legendofmiracles — COMMENTED** [review]
PR title should be `nestub: init at 1.20, nixos/nesthub: init`

File: nixos/modules/services/misc/nesthub.nix (hunk 2 above)
Comment (bobby285271) [reply nixos/modules/services/misc/nesthub.nix:14]: https://github.com/NixOS/nixpkgs/blob/780f3cf9facc320dba8709ff4c4caf9f3f94aded/lib/options.nix#L93-L100

We probably don't need the `Whether to enable` prefix and the period if we use `mkEnableOption`.

**Pull request closed without merging**